assets/app/scripts/directives/nav.js

```diff
@@ -20,86 +20,78 @@ angular.module('openshiftConsole')
       templateUrl: "views/_sidebar-main-nav-item.html"
     };
   })
-  .directive('projectNav', function($timeout, $location, $filter, LabelFilter) {
+  .directive('projectNav', function($timeout, $location, $filter, LabelFilter, DataService) {
     return {
-      restrict: 'E',
+      restrict: 'EA',
       templateUrl: 'views/_project-nav.html',
-      link: function ($scope, element) {
-        var select = $('.selectpicker', element);
+      link: function($scope, $elem) {
+        var select = $elem.find('.selectpicker');
+        var projects = {};
+        var sortedProjects = [];
+        var options = [];
```

Comment:
I'd kind of like to break this down into a few smaller directives that would look something like this:

```html
<project-nav>
  <projects-list></projects-list>
  <projects-filter></projects-filter>
  <button>Add to Project</button>
</project-nav>
```
then isolate their `$scopes` and let them talk via api by using `require` to grab ahold of controllers.   Otherwise, I'm ok with the link function the way it is since it isn't providing any communication.  Moving the request to a controller may speed it up minimally, but I'm not sure if it would be worth the extra complexity.